06_gpu_and_ml/embeddings/infinity/embedding_racetrack.py

```diff
@@ -0,0 +1,213 @@
+
+# # Modal Cookbook: Recipe for Inference Throughput Maximization
+# In certain applications, the bottom line comes to throughput: process a set of inputs as fast as possible.
+# Let's explore how to maximize throughput by using Modal on an embedding example, and see just how fast
+# we can encode 10,000 images from the [wildflow sweet-coral dataset](https://huggingface.co/datasets/wildflow/sweet-corals "huggingface/wildflow/sweet-coral")
+# using the [Infinity inference engine](https://github.com/michaelfeil/infinity "github/michaelfeil/infinity").
+
+# ## BLUF (bottom line up front)
+# We have found that setting concurrency to 2 and then maximizing the batchsize will maximize GPU utilization.
+# If you then enable more containers, throughput goes through the roof.
+
+## Setup
+# Import everything we need for the locally-run Python (everything in our local_entrypoint function at the bottom).
+from time import perf_counter
+from pathlib import Path
+from more_itertools import chunked
+import csv, os
+import numpy as np
+
+import modal
+from modal.volume import FileEntry
+from PIL.Image import Image
+
+# We will be running this example several times, and logging speeds to a local file specified here.
+# Set local_logfile to `None` to ignore this.
+local_logfile = None # Path("~/results/rt6.txt").expanduser()
+
+app_name = 'embedding-racetrack'
```

Comment:
Variable isn't used anywhere. Let's drop it. Also, for convention our examples start with `example-`.